hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/metrics/RBFMetrics.java

```diff
@@ -712,13 +715,22 @@ public long getCurrentTokensCount() {
 
   @Override
   public String getTopTokenRealOwners() {
```

Comment:
Do we have a unit test for this?

Reply:
We don't have one. I had added 
`@Test
  public void testGetTopTokenRealOwners() throws Exception {
    RouterMBean bean = FederationTestUtils.getBean(
            ROUTER_BEAN, RouterMBean.class);
    String output = bean.getTopTokenRealOwners();
    assertNotNull("The value of output should not be null " + output, output);
    assertTrue("The value contains incorrect message " + output, !output.contains("NameValuePair"));
  }` in TestRouterHDFSContractDelegationToken (A similar function in RBFMetrics was being called in this test class) to test the patch in local

Reply:
@NishthaShah Where is this test added, can you add a test in the PR as well?

Reply:
Sure @ayushtkn, Let me add it in the same class (TestRouterHDFSContractDelegationToken) where I added to test

Reply:
@ayushtkn Let me know if adding it in TestRouterHDFSContractDelegationToken, sounds good or we should add it in some more suitable place. (Initially had tried to add a test in TestRBFMetrics, but this.router.getRpcServer(), is failing with NullPointerException)

Reply:
TestRouterHDFSContractDelegationToken contract test aren't a good place, can explore ``TestRouterSecurityManager``

Reply:
@ayushtkn Moved and tested by adding in TestRouterSecurityManager